django_project/minisass_frontend/src/pages/MainPage/index.tsx

```diff
@@ -27,7 +27,8 @@ const Home: React.FC = () => {
   const ObservationsPropList = [];
   const [activationComplete, setActivationComplete] = useState(false);
   const [activationMessage, setActivationMessage] = useState('');
-  const [observationPerPage, setObservationPerPage] = useState(5)
+  const [observationPerPage, setObservationPerPage] = useState(5);
+  const [retryCount, setRetryCount] = useState(0);
```

Comment:
@tinashechiraya 
You don't need to save it as state
It will make re-render everything that is not needed

Just put the retry count as parameter 

```
const fetchHomePageData = (retryCount = 0) => {
```
```
fetchHomePageData(retryCount+1);
```